pymove/tests/preprocessing/test_compression.py

```diff
@@ -0,0 +1,78 @@
+from numpy import nan
+from pandas import DataFrame, Timestamp
+from pandas.testing import assert_frame_equal
+
+from pymove import MoveDataFrame, compression
+
+list_data = [
+    [39.984093, 116.319237, '2008-10-23 05:53:05', 1],
+    [39.984200, 116.319321, '2008-10-23 05:53:06', 1],
+    [39.984222, 116.319405, '2008-10-23 05:53:11', 1],
+    [39.984211, 116.319389, '2008-10-23 05:53:16', 1],
+    [39.984219, 116.319420, '2008-10-23 05:53:21', 1]
+]
+
+
+def test_compress_segment_to_stop_point():
+    move_df = MoveDataFrame(
+        data=list_data, latitude='lat', longitude='lon',
+        datetime='datetime', traj_id='id'
+    )
+    move_df['segment_stop'] = [1, 2, 3, 3, 3]
+    move_df['dist_to_prev'] = [nan, nan, nan, 1.8315003203265166, 2.786977605326357]
+    move_df['time_to_prev'] = [nan, nan, nan, 5.0, 5.0]
+    move_df['speed_to_prev'] = [nan, nan, nan, 0.3663000640653033, 0.5573955210652713]
+    move_df['stop'] = [False, False, True, True, True]
+    cols = [
+        'lat', 'lon', 'datetime', 'id',
+        'segment_stop', 'dist_to_prev', 'time_to_prev', 'speed_to_prev',
+        'stop', 'lat_mean', 'lon_mean'
+    ]
+
+    compressed_trajs_mean = compression.compress_segment_stop_to_point(
+        move_df, dist_radius=3.5, time_radius=0.5
+    )
+    expected = DataFrame(data=[
+        [39.984093, 116.319237, Timestamp('2008-10-23 05:53:05'), 1, 1,
+         nan, nan, nan, False, nan, nan],
+        [39.9842, 116.319321, Timestamp('2008-10-23 05:53:06'), 1, 2,
+         nan, nan, nan, False, nan, nan],
+        [39.984222, 116.319405, Timestamp('2008-10-23 05:53:11'), 1, 3,
+         nan, nan, nan, True, 39.984222, 116.319405],
+        [39.984219, 116.31942, Timestamp('2008-10-23 05:53:21'), 1, 3,
+         2.786977605326357, 5.0, 0.5573955210652713, True, 39.984222, 116.319405],
+    ], columns=cols, index=[0, 1, 2, 4]
+    )
+    assert_frame_equal(compressed_trajs_mean, expected)
+    assert move_df.len() == 5
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5574862)